docs/reference/modules/cluster/remote-clusters-migration.asciidoc

```diff
@@ -0,0 +1,242 @@
+==== Reconfigure the remote cluster and generate a cross-cluster API key
+
+On the remote cluster:
+
+include::remote-clusters-api-key.asciidoc[tag=remote-cluster-steps]
+
+[[remote-clusters-migration-stop]]
+==== Stop cross-cluster operations
+
+On the local cluster, stop any persistent tasks that refer to the remote 
+cluster:
+
+* Use the <<stop-transform>> API to stop any transforms.
+* Use the <<ml-close-job>> API to close any anomaly detection jobs.
+* Use the <<ccr-pause-auto-follow-pattern>> API to pause any auto-follow {ccr}.
+* Use the <<ccr-post-pause-follow>> API to pause any manual {ccr} or existing
+indices that were created from the auto-follow pattern.
+
+[[remote-clusters-migration-reconnect]]
+==== Reconnect to the remote cluster
+
+On the local cluster:
+
+. Enhance any roles used by local cluster users with the required
+<<roles-remote-indices-priv,remote indices privileges>> for {ccr} and {ccs}.
+Refer to <<remote-clusters-privileges-api-key>>. Note:
+** You only need to assign additional `remote_indices` privileges to existing
+roles.
+** No update is needed if the {ccr} or {ccs} tasks have been configured with a
+`superuser` role. The `superuser` role is automatically updated to allow access
+to all remote indices.
+** Tasks that are run as regular users with named roles are immediately updated
+with the new privileges. A task will load a new definition the next time it 
+runs.
+** You need to restart tasks that are run using an API key (done in a later
+step).
+
+. If you've dynamically configured the remote cluster (via the cluster settings
+API):
+
+.. Retrieve the current remote cluster configuration, and store it in a safe
+place. You may need it later in case you need to
+<<remote-clusters-migration-rollback,roll back>>. Use the cluster settings API:
++
+[source,console]
+----
+GET /_cluster/settings?filter_path=persistent.cluster.remote
+----
+
+.. Remove the existing remote cluster definition by setting the remote cluster
+settings to `null`.
+
+. If you've statically configured the remote cluster (via `elasticsearch.yml`),
+copy the `cluster.remote` settings from `elasticsearch.yml`, and store them
+in a safe place. You may need them later in case you need to
+<<remote-clusters-migration-rollback,roll back>>.
+
+include::remote-clusters-api-key.asciidoc[tag=local-cluster-steps]
+
+. If you've dynamically configured the remote cluster (via the cluster settings
+API):
+
+.. Restart the local cluster to load changes to the keystore.
+
+.. Re-add the remote cluster. Use the same remote cluster alias, and change the
+transport port into the remote cluster port. For example:
++
+[source,console]
+----
+PUT /_cluster/settings
+{
+  "persistent" : {
+    "cluster" : {
+      "remote" : {
+        "my_remote" : { <1>
+          "mode": "proxy",
+          "proxy_address": "my.remote.cluster.com:9443" <2>
+        }
+      }
+    }
+  }
+}
+----
+// TEST[skip:TODO]
+<1> The remote cluster alias.
```

Comment:
I think we probably want to emphasize this should be the same as the existing alias for in-place update.

Reply:
This made me realize that we should also emphasize this earlier, when adding the key to the local keystore with `./bin/elasticsearch-keystore add cluster.remote.ALIAS.credentials`. I've added that clarification in https://github.com/elastic/elasticsearch/pull/98999/commits/ea4584c30623f194ca88a1098fd3b27886030e68